src/protobuf.zig

```diff
@@ -100,6 +108,45 @@ pub const ManagedString = union(ManagedStringTag) {
     }
 };
 
+pub const ManagedStructTag = enum {
+    Owned,
+    Managed
+};
+
+pub fn ManagedStruct(T: type) type {
+    return union(ManagedStructTag) {
+        Owned: AllocatedStruct(T),
+        Managed: *const T,
+
+        const Self = @This();
+
+        pub fn managed(p : *const T) ManagedStruct(T){
+            return ManagedStruct(T) {.Managed = p};
+        }
+
+        pub fn move(p: *T, allocator : Allocator) ManagedStruct(T) {
```

Comment:
Probably a doc comment about who owns the memory and should be responsible for the deallocation would be nice here